res/skins/QMLDemo/OrientationToggleButton.qml

```diff
@@ -2,79 +2,60 @@ import "." as Skin
 import Mixxx 0.1 as Mixxx
 import QtQuick 2.12
 import QtQuick.Controls 2.12
+import "util.mjs" as Util
 
-AbstractButton {
+Item {
     id: root
 
     property string group // required
     property string key // required
-    property alias orientation: control.value
+    property alias orientation: orientationSlider.value
     property color color: "white"
 
     implicitWidth: 56
     implicitHeight: 26
-    onClicked: control.value = Math.trunc(3 * pressX / root.width)
-    states: [
-        State {
-            name: "left"
-            when: orientation == 0
-
-            PropertyChanges {
-                target: indicator
-                x: 0
-            }
-
-        },
-        State {
-            name: "right"
-            when: orientation == 2
-
-            PropertyChanges {
-                target: indicator
-                x: parent.width - width
-            }
 
-        },
-        State {
-            name: "mid"
-            when: orientation == 1
+    Slider {
+        id: orientationSlider
 
-            PropertyChanges {
-                target: indicator
-                x: parent.width / 2 - width / 2
-            }
-
-        }
-    ]
-
-    Mixxx.ControlProxy {
-        id: control
-
-        group: root.group
-        key: root.key
-    }
-
-    Item {
-        anchors.fill: root
+        anchors.fill: parent
         anchors.leftMargin: 10
         anchors.rightMargin: 10
         anchors.topMargin: 2
         anchors.bottomMargin: 2
+        wheelEnabled: false
+        live: false
+        from: 0
+        to: 2
+        stepSize: 1
+        value: control.value
+        orientation: Qt.Horizontal
+        snapMode: Slider.SnapOnRelease
+        onMoved: {
+            // The slider's `value` is not updated until after the move ended.
+            const val = valueAt(visualPosition);
+            if (Util.hasChanged(val))
```

Comment:
Why not just `val != control.value`? This utility function makes it harder to understand what happens. It's also not reusable because it uses a global mutable variable (evil!).

Reply:
Thats a good idea actually. How did I not think of that.